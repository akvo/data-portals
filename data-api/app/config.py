from typing import Any, Dict, Optional

from pydantic import BaseSettings, PostgresDsn, validator


class Settings(BaseSettings):
    POSTGRES_HOST: str
    POSTGRES_USER: str
    POSTGRES_PASSWORD: str
    POSTGRES_DB: str
    DATABASE_URI: Optional[PostgresDsn] = None

    @validator("DATABASE_URI", pre=True)
    def assemble_db_connection(cls, v: Optional[str], values: Dict[str, Any]) -> Any:
        if isinstance(v, str):  # pragma: no cover
            return v
        return PostgresDsn.build(
            scheme="postgresql",
            user=values.get("POSTGRES_USER"),
            password=values.get("POSTGRES_PASSWORD"),
            host=values.get("POSTGRES_HOST"),
            path=f"/{values.get('POSTGRES_DB') or ''}",
        )
