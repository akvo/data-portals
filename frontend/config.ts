export const DOMAIN = process.env.DOMAIN || ''

export const INTERNAL_API_DOMAIN = process.env.INTERNAL_API_DOMAIN || ''

export const API_PATH = process.env.API_PATH || ''

export const API_URL = DOMAIN + API_PATH

export const INTERNAL_API_URL = INTERNAL_API_DOMAIN + API_PATH

export const MAPBOX_TOKEN = process.env.MAPBOX_TOKEN || ''
