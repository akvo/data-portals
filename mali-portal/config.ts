export const API_HOST = process.env.API_HOST

export const DATA_ENDPOINT = process.env.DATA_ENDPOINT

export const DATA_API_ENDPOINT = process.env.DATA_API_ENDPOINT

export const DEFAULT_MAP_VIEWPORT = {
  latitude: 17.65,
  longitude: -4.15,
  zoom: 4.4,
}
