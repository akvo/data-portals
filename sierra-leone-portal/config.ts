export const API_HOST = process.env.API_HOST

export const DATA_ENDPOINT = process.env.DATA_ENDPOINT

export const DATA_API_ENDPOINT = process.env.DATA_API_ENDPOINT

export const DEFAULT_MAP_VIEWPORT = {
  latitude: 8.4,
  longitude: -11.8,
  zoom: 6.5,
}
