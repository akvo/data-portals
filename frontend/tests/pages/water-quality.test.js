import React from 'react'
import { render } from '@testing-library/react'
import WaterQuality, { getServerSideProps } from '../../pages/water-quality'
import { server, rest } from '../mocks/server'
import { API_PATH } from '../../config'

describe('Water Quality page', () => {
  const fakeGeojson = {
    type: 'FeatureCollection',
    features: [],
  }

  it('matches snapshot', () => {
    const sourceUrl = API_PATH + '/site-points.geojson'

    server.use(
      rest.get(sourceUrl, (_, res, ctx) => {
        return res(ctx.json(fakeGeojson))
      })
    )

    const { asFragment } = render(<WaterQuality sourceUrl={sourceUrl} />, {})

    expect(asFragment()).toMatchSnapshot()
  })

  test('getServerSideProps', async () => {
    const respone = await getServerSideProps()

    expect(respone.props).toHaveProperty('sourceUrl')
    expect(respone.props).toHaveProperty('mapboxToken')
  })
})
