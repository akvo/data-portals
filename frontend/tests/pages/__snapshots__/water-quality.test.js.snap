// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Water Quality page matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Water quality
  </h1>
  <p>
    Ullamcorper dignissim cras tincidunt lobortis. Est lorem ipsum dolor sit amet consectetur adipiscing. Tempor commodo ullamcorper a lacus. Morbi tristique senectus et netus et. Posuere ac ut consequat semper viverra nam libero. Massa vitae tortor condimentum lacinia quis vel eros donec ac. Ultricies integer quis auctor elit sed vulputate mi sit amet. Vitae congue eu consequat ac felis. Vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras. Tortor at risus viverra adipiscing. Neque volutpat ac tincidunt vitae semper quis lectus nulla at. Urna id volutpat lacus laoreet non. Venenatis tellus in metus vulputate eu.
  </p>
  <p>
    Sapien pellentesque habitant morbi tristique senectus. Senectus et netus et malesuada fames ac turpis. Dignissim diam quis enim lobortis scelerisque. Praesent elementum facilisis leo vel fringilla est ullamcorper. Pellentesque elit eget gravida cum sociis natoque penatibus et magnis. Facilisis gravida neque convallis a cras semper auctor neque vitae. Netus et malesuada fames ac. Tortor condimentum lacinia quis vel eros. Non quam lacus suspendisse faucibus interdum posuere lorem ipsum. Urna molestie at elementum eu facilisis sed odio morbi. Urna et pharetra pharetra massa massa. Nec ultrices dui sapien eget mi proin sed. Aenean sed adipiscing diam donec adipiscing tristique risus nec feugiat. Facilisi morbi tempus iaculis urna id volutpat lacus laoreet. Dolor sit amet consectetur adipiscing. Amet commodo nulla facilisi nullam vehicula ipsum a arcu. Eu augue ut lectus arcu bibendum at varius vel pharetra. Sed adipiscing diam donec adipiscing tristique risus nec.
  </p>
  <div
    style="height: 50vh; width: 100%; max-width: 1000px;"
  >
    <div
      style="position: relative; width: 100%; height: 100%; cursor: grab; outline: none;"
      tabindex="0"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      >
        <div
          class=""
          style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
        />
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        />
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
